cirq-core/cirq/ops/gate_operation.py

```diff
@@ -146,17 +146,13 @@ def _group_interchangeable_qubits(
     ) -> Tuple[Union['cirq.Qid', Tuple[int, FrozenSet['cirq.Qid']]], ...]:
         if not isinstance(self.gate, gate_features.InterchangeableQubitsGate):
             return self.qubits
-        else:
-
-            def make_key(i_q: Tuple[int, 'cirq.Qid']) -> int:
-                return cast(
-                    gate_features.InterchangeableQubitsGate, self.gate
-                ).qubit_index_to_equivalence_group_key(i_q[0])
-
-            return tuple(
-                (k, frozenset(g for _, g in kg))
-                for k, kg in itertools.groupby(enumerate(self.qubits), make_key)
-            )
+        groups: Dict[int, List['cirq.Qid']] = {}
+        for i, q in enumerate(self.qubits):
+            k = self.gate.qubit_index_to_equivalence_group_key(i)
+            if k not in groups:
+                groups[k] = []
+            groups[k].append(q)
```

Comment:
Can combine these lines using `setdefault`:
```suggestion
            groups.setdefault(k, []).append(q)
```